taskvine/src/manager/vine_manager.c

```diff
@@ -2506,21 +2513,15 @@ struct rmsummary *vine_manager_choose_resources_for_task(
 			max_proportion = MAX(max_proportion, limits->gpus / w->resources->gpus.largest);
 		}
 
-		// if max_proportion > 1, then the task does not fit the worker for the
-		// specified resources. For the unspecified resources we use the whole
-		// worker as not to trigger a warning when checking for tasks that can't
-		// run on any available worker.
+		/* If max_proportion > 1, then the task does not fit the worker for the
+		 * specified resources. For the unspecified resources we use the whole
+		 * worker as not to trigger a warning when checking for tasks that can't
+		 * run on any available worker. */
 		if (max_proportion > 1) {
 			use_whole_worker = 1;
 		} else if (max_proportion > 0) {
 			use_whole_worker = 0;
 
-			// adjust max_proportion so that an integer number of tasks fit the
```

Comment:
Hmm, I think this is an important step for normal tasks.
Maybe make this conditional on `!t->needs_library` so that it doesn't apply to functions?